src/block_cache/block_chunkcache.c

```diff
@@ -20,6 +20,43 @@
 #define WT_CHUNK_OFFSET(chunkcache, offset) \
     (wt_off_t)(((size_t)offset / (chunkcache)->chunk_size) * (chunkcache)->chunk_size)
 
+/*
+ * __bitmap_find_free --
+ *     Iterate through the bitmap to find a free chunk in the cache.
+ */
+static int
+__bitmap_find_free(WT_SESSION_IMPL *session, size_t *bit_index)
+{
+    WT_CHUNKCACHE *chunkcache;
+    size_t bitmap_size, bit_remainder;
+    uint8_t map_byte;
+
+    chunkcache = &S2C(session)->chunkcache;
+    bitmap_size = chunkcache->capacity / chunkcache->chunk_size;
+
+    /* Iterate through the bytes and bits of the bitmap to find free chunks */
+    for (size_t i = 0; i < bitmap_size; i++) {
+        map_byte = chunkcache->bitmap[i];
+        if (map_byte != 0xff) {
+            for (size_t j = 0; j < 8; j++) {
```

Comment:
```suggestion
            size_t j = 0;
            /* shift right until we find the first zero */
            while ((map_byte & 1) != 0) {
                j++;
                map_byte >>= 1;
            }
```
The suggested code is for *reference*: it can be reformulated to utilize a `for` loop.

The logic is a little more compact and by shifting the value to right it avoids the need for an explicit mask. Another benefit is by shifting to the right, it completely avoids the possibility of a static checker (like Coverity) warning that the shift count may exceed type width.

Reply:
Thanks for the suggestion Marc, this is a clean implementation and works the same, I don't see why not to implement it. I've done so here [1b521a5].(https://github.com/wiredtiger/wiredtiger/pull/9442/commits/1b521a5b610b85741d3c0d3c348c7f0bf74d2846)

Reply:
I'm glad it was of use.